resources/prosody-plugins/mod_muc_room_metadata.lua

```diff
@@ -0,0 +1,237 @@
+-- This module implements a generic metadata storage system for rooms.
+--
+-- This module is added under the main virtual host domain
+-- It needs a room metadata muc component
+--
+-- VirtualHost "jitmeet.example.com"
+--     modules_enabled = {
+--         "muc_room_metadata"
+--     }
+--     room_metadata_muc = "metadata.jitmeet.example.com"
+--     main_muc = "muc.jitmeet.example.com"
+--
+-- Component "metadata.jitmeet.example.com" "muc"
```

Comment:
Why do we need this to be a muc? That just gets messages and should be like speaker stats component, does not need to be a muc component, right?